test/e2e/framework/fixture.go

```diff
@@ -271,89 +271,187 @@ func NewWorkspaceWithWorkloads(t *testing.T, server RunningServer, orgClusterNam
+	syncerNamespace := workloadcliplugin.GetSyncerID(sf.WorkspaceClusterName.String(), sf.WorkloadClusterName)
 	syncerConfig := syncerConfigFromCluster(t, downstreamConfig, syncerNamespace)
 
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	t.Cleanup(cancelFunc)
+
 	downstreamKubeClient, err := kubernetesclientset.NewForConfig(downstreamConfig)
 	require.NoError(t, err)
 
-	return &SyncerFixture{
-		RunningServer:        downstreamServer,
-		DownstreamKubeClient: downstreamKubeClient,
+	if useDeployedSyncer {
+		// Ensure cleanup of pcluster resources
+
+		syncerID := syncerConfig.ID()
+		t.Cleanup(func() {
+			t.Logf("Deleting syncer resources for logical cluster %q, workload cluster %q", syncerConfig.KCPClusterName, syncerConfig.WorkloadClusterName)
+			err := downstreamKubeClient.CoreV1().Namespaces().Delete(ctx, syncerID, metav1.DeleteOptions{})
+			if err != nil {
+				t.Errorf("failed to delete Namespace %q: %v", syncerID, err)
+			}
+			err = downstreamKubeClient.RbacV1().ClusterRoleBindings().Delete(ctx, syncerID, metav1.DeleteOptions{})
+			if err != nil {
+				t.Errorf("failed to delete ClusterRoleBinding %q: %v", syncerID, err)
+			}
+			err = downstreamKubeClient.RbacV1().ClusterRoles().Delete(ctx, syncerID, metav1.DeleteOptions{})
+			if err != nil {
+				t.Errorf("failed to delete ClusterRole %q: %v", syncerID, err)
+			}
+
+			t.Logf("Deleting synced resources for logical cluster %q, workload cluster %q", syncerConfig.KCPClusterName, syncerConfig.WorkloadClusterName)
+			namespaces, err := downstreamKubeClient.CoreV1().Namespaces().List(ctx, metav1.ListOptions{})
+			if err != nil {
+				t.Errorf("failed to list namespaces: %v", err)
+			}
+			for _, ns := range namespaces.Items {
+				locator, err := syncer.LocatorFromAnnotations(ns.Annotations)
+				if err != nil {
+					t.Errorf("failed to retrieve locator from ns %q: %v", ns.Name, err)
+					continue
+				}
+				if locator == nil {
+					// Not a kcp-synced namespace
+					continue
+				}
+				if locator.LogicalCluster != syncerConfig.KCPClusterName {
+					// Not a namespace synced by this syncer
+					continue
+				}
+				err = downstreamKubeClient.CoreV1().Namespaces().Delete(ctx, ns.Name, metav1.DeleteOptions{})
+				if err != nil {
+					t.Errorf("failed to delete Namespace %q: %v", ns.Name, err)
+				}
+			}
+		})
+
+	} else {
+		// Start an in-process syncer
+
+		err := syncer.StartSyncer(ctx, syncerConfig, 2, 5*time.Second)
+		require.NoError(t, err, "syncer failed to start")
+	}
+
+	startedSyncer := &StartedSyncerFixture{
 		SyncerConfig:         syncerConfig,
+		DownstreamConfig:     downstreamConfig,
+		DownstreamKubeClient: downstreamKubeClient,
 	}
+
+	// The workload cluster becoming ready indicates the syncer is healthy and has
+	// successfully sent a heartbeat to kcp.
+	startedSyncer.WaitForClusterReadyReason(t, ctx, "")
+
+	return startedSyncer
+}
+
+// StartedSyncerFixture contains the configuration used to start a syncer and interact with its
```

Comment:
Are we still using this to start a syncer?

Reply:
'Started' is the key word here - it results from starting the fixture. If you don't like it, easy enough to change, but I thought it was useful to allow configuring the fixture from a struct (to allow not specifying optional things) and then start it subsequently.